mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -218,12 +233,14 @@ bool MemRefDependenceGraph::init() {
     unsigned n = memrefAndList.second.size();
     for (unsigned i = 0; i < n; ++i) {
       unsigned srcId = memrefAndList.second[i];
-      bool srcHasStore =
-          getNode(srcId)->getStoreOpCount(memrefAndList.first) > 0;
+      Node *srcNode = getNode(srcId);
```

Comment:
An edge is introduced between the srcNode and dstNode if either has a memory effect or a store op.
The current algorithm tracks memoryEffectOps which are within a loop body.  Top-level ops with memoryEffect are handled separately for fusion validity checks.